File: BiblioTec/DTOs/CategoriaDto.cs
namespace BiblioTec.DTOs.Categorias
{
    public class CategoriaDto
    {
        public int IdCategoria { get; set; }
        public string Nombre { get; set; }
    }

    public class CategoriaCreateDto
    {
        public string Nombre { get; set; }
    }
    public class LibroCategoriaDto
    {
        public List<int> Categorias { get; set; } = new();
    }
}

File: BiblioTecAdmin/src/main/java/com/bibliotec/admin/model/CategoriaDto.java
package com.bibliotec.admin.model;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class CategoriaDto {
    public int idCategoria;
    public String nombre;

    public int    getIdCategoria() { return idCategoria; }
    public String getNombre()      { return nombre != null ? nombre : ""; }

    @Override public String toString() { return nombre != null ? nombre : ""; }
}
